pkg/controller/instance/resolver_incluster.go

```diff
@@ -9,9 +9,16 @@ import (
 	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
 )
 
-// InClusterResolver resolves packages that are already installed in the cluster. Note, that unlike other resolvers, the
-// resulting 'packages.Package' struct does not contain package 'packages.Files' (we don't have the original files) and
-// doesn't have an Instance resource because multiple Instances of the same Operator/OperatorVersion can exist
+// InClusterResolver is a server-side package resolver for packages that are already installed in the cluster. It is a simpler
+// version of the client-side pkg/kudoctl/packages/resolver/resolver_incluster.go. The client-side version would search
+// the installed OperatorVersions and try to resolve any valid combination of the operator name and its app and operator versions,
+// same as we would search in the repository.
+// This resolver is only used to make sure that all the dependencies of an operator exist and that referenced operator versions
+// are installed and uniquely identifiable by the passed operator name, appVersion and operatorVersion parameters
+// (see pkg/apis/kudo/v1beta1/operatorversion_types_helpers.go::OperatorVersionName method).
+// Also, note that unlike other resolvers, the resulting 'packages.Package' struct does not contain package 'packages.Files'
```

Comment:
when you need to write a comment like this to a type I think it already means you're doing something wrong :)

Reply:
That's an already existing **server-side** in-cluster resolver. don't mind it - I just updated the docs.